File: starter-projects/typescript-expressjs-api-mongodb-mongoose/src/model/post.ts
import { Schema, model } from 'mongoose';

type Post = {
  title: string;
  content: string;
  authorEmail: string;
};

const PostSchema = new Schema<Post>(
  {
    title: {
      type: String,
      required: true,
    },
    content: {
      type: String,
      required: true,
    },
    authorEmail: {
      type: String,
      required: true,
    },
  },
  { timestamps: true }
);

export const Post = model<Post>('Post', PostSchema);

File: starter-projects/phoenix-api-postgres/lib/phoenix_api/post.ex
defmodule PhoenixApi.Post do
  use Ecto.Schema
  import Ecto.Changeset

  @primary_key {:id, :binary_id, autogenerate: true}
  schema "posts" do
    field :title, :string
    field :content, :string
    field :author_email, :string
    timestamps()
  end

  def changeset(post, attrs) do
    post
    |> cast(attrs, [:title, :content, :author_email])
    |> validate_required([:title])
  end
end
